fractal_server/app/models/workflow.py

```diff
@@ -94,3 +94,24 @@ async def insert_task(
         if commit:
             await db.commit()
         return wf_task
+
+    async def remove_task(
+        self,
+        task_id: int,
+        *,
+        args: Dict[str, Any] = None,
+        db: AsyncSession,
+        commit: bool = True,
+    ) -> WorkflowTask:
+
+        for i, wftask in enumerate(self.task_list):
+            if wftask.task.id == task_id:
+                j = i + 1
+                for _wftask in self.task_list[j:]:
+                    _wftask.order -= 1
+                    await db.merge(_wftask)
+                await db.delete(wftask)
+                self.task_list.remove(wftask)
+                await db.commit()
+                break
+        return
```

Comment:
deletion should always take place by WorkflowTask.id, which is the primary key and can thus be deleted in a one liner. please remove this method